src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -714,7 +713,7 @@ public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependenci
 
         protected override void OnMarked(NodeFactory context)
         {
-            Debug.Assert(_type.IsTypeDefinition || !_type.HasSameTypeDefinition(context.ArrayOfTClass), "Asking for Array<T> EEType");
+            //Debug.Assert(_type.IsTypeDefinition || !_type.HasSameTypeDefinition(context.ArrayOfTClass), "Asking for Array<T> EEType");
```

Comment:
We'll need to find a way to make it work with this assert in place. We really shouldn't be seeing EETypes for `Array<T>` instances being generated.